osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs

```diff
@@ -39,10 +40,12 @@ protected DifficultyCalculator(Ruleset ruleset, WorkingBeatmap beatmap)
         /// Calculates the difficulty of the beatmap using a specific mod combination.
         /// </summary>
         /// <param name="mods">The mods that should be applied to the beatmap.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>A structure describing the difficulty of the beatmap.</returns>
-        public DifficultyAttributes Calculate(params Mod[] mods)
+        public DifficultyAttributes Calculate(IEnumerable<Mod> mods = null, CancellationToken cancellationToken = default)
```

Comment:
No idea why these mod changes were made...

Reply:
Can't have `params` and optional parameters with defaults in one method, I don't think.

Reply:
I see. I've changed this to an overload method, I think it feels better?